trpcultivate_phenotypes/src/Service/TripalCultivatePhenotypesGenusProjectService.php

```diff
@@ -0,0 +1,154 @@
+    $this->chado = $chado;
+
+    // Tripal Logger service.
+    $this->logger = $logger;
+  }
+  
+  /**
+   * Assign a genus to an experiment/project.
+   * 
+   * @param int $project
+   *   Project (project id number) the parameter $genus will be assigned to.
+   * @param string $genus
+   *   Genus name/title.
+   * @param boolean $replace
+   *   True to replace existing genus of a project with a different genus.
+   *   Default to False.
+   *
+   * @return boolean
+   *   True, genus was set successfully or false on error/fail.
+   */
+  function setGenusToProject($project, $genus, $replace = FALSE) {
+    $error = 0;
+
+    if ($project > 0 && !empty($genus)) {
+      $result = $this->chado->query("
+        SELECT projectprop_id AS id FROM {1:projectprop} 
+        WHERE project_id = :project_id AND type_id = :type_id LIMIT 1
+      ", [':project_id' => $project, ':type_id' => $this->sysvar_genus]);
+      
+      $projectprop_id = $result->fetchField();
+
+      if ($projectprop_id > 0) {
+        // Has a genus.
+        if ($replace) {
+          // And wishes to replace with another genus.
+          $this->chado->query("
+            UPDATE {1:projectprop} SET value = :new_genus WHERE projectprop_id = :id
+          ", [':new_genus' => $genus, ':id' => $projectprop_id]);
+        }
+
+        // Do nothing if maintain the same genus.
+      } 
+      else {
+        // Not set yet, no record in projectprop.
+        // Create a relationship regardless to replace or not.
+        $sql = "INSERT INTO {1:projectprop} (project_id, type_id, value) VALUES ('%s', '%s', '%s')";
+        $query = sprintf($sql, $project, $this->sysvar_genus, $genus);
+        $this->chado->query($query);
+      }
+    }
+    else {
+      $error = 1;
+      $this->logger->error('Error. Could not replace genus. Invalid project or genus');
+    }
+
+    return ($error) ? FALSE : TRUE;
+  }
+
+  /**
+   * Get genus of an experiment/project.
+   * 
+   * @param int $project
+   *   Project (project_id number) to search.
+   *
+   * @return array
+   *   Key is genus/organism id number and value is the genus name/title.    
+   */
+  public function getGenusOfProject($project) {
+    $genus_project = 0;
+
+    if ($project > 0) {
+      $result = $this->chado->query("
+        SELECT organism_id AS id, genus FROM {1:organism} 
+        WHERE genus = (SELECT value::VARCHAR FROM {1:projectprop} 
+          WHERE project_id = :project_id AND type_id = :type_id LIMIT 1)
+        LIMIT 1
+      ", [':project_id' => $project, ':type_id' => $this->sysvar_genus]);
+
+      $genus_project = $result->fetchObject();
+    }
+
+    return ($genus_project) ? ['id' => $genus_project->id, 'genus' => $genus_project->genus] : 0;
+  }
+
+  /**
+   * Get all genus that have been configured (traits, unit, method, database and crop ontology).
+   *
+   * @return array
+   *   An array of genus names, sorted alphabetically.
+   */
+  public function getActiveGenus() {
```

Comment:
makes perfect sense, I have evaluated the purpose of the method and I think that it can be omitted since it will be used mainly for populating select field and as discussed and agreed that for populating select field we could just straight up write a query or if Tripal 4 provides.